File: Data Structure/Lab 24/LinearSearch.c
// Write a program to implement a Linear Search using Array.

#include<stdio.h>

int LinearSearch(int arr[], int size, int x){

    for(int i = 0; i < size; i++){
        if(arr[i] == x){
            return i;
        }
    }
    return -1;
}

void main(){
    int n , x , index;

    printf("Enter size of an array:");
    scanf("%d", &n);

    int arr[n];

    printf("Enter array elements:");
    for(int i = 0; i < n ; i++){
        scanf("%d", &arr[i]);
    }

    printf("Enter value to find:");
    scanf("%d", &x);

    index = LinearSearch(arr , n , x);

    printf("Value found at : %d", index);
}

File: Data Structure(Java)/Lab 22/LinearSearch.java
// Write a program to implement a Linear Search using Array.

import java.util.Scanner;

public class LinearSearch {

    static int Linearsearch(int arr[], int size, int x){
        for(int i = 0 ; i < size ; i++){
            if(arr[i] == x){
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter size of an array:");
        int n = sc.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter array elements:");
        for(int i = 0 ; i < n ; i++){
            arr[i] = sc.nextInt();
        }

        System.out.println("Enter value to find:");
        int x = sc.nextInt();

        int index = Linearsearch(arr, n, x);

        System.out.println("Value found at index :"+ index);

        sc.close();
    }
}
